File: data-structures/list/problems/ts/leetcode-27.ts
function removeElement(nums: number[], val: number): number {
    let count = 0;
    
    for(let i = 0; i < nums.length; i++) { // [0,1,2,2,3,0,4,2], 2
        if(nums[i] === val) {
            nums[i] = NaN
        } else {
            count++
        }
    }

    nums.sort()

    return count
};

File: data-structures/list/problems/go/leetcode-27.go
package problems

var nums = []int{3, 2, 2, 3}
var val = 3

func removeElement(nums []int, val int) int {
	p := 0
	for i := range nums {
		if nums[i] != val {
			nums[p] = nums[i]
			p++
		}
	}
	return p
}
